themes/hugo-steam-theme/static/js/index.js

```diff
@@ -1,18 +1,59 @@
-/**
- * Main JS file for Casper behaviours
- */
+(function($) {
+  var $forms = $('.slack-form');
+  var emailRegex = /^.+@.+\..+$/;
 
-/*globals jQuery, document */
-(function ($) {
-    "use strict";
+  function invite(url, email) {
+    return $.ajax({
+      url: url,
+      method: 'POST',
+      dataType: 'json',
+      data: JSON.stringify({
+        email: email
+      })
+    });
+  }
+
+  function setupForm() {
+    var $this = $(this);
+    var $form = $this.find('form');
+    var $input = $this.find('input');
+    var $btn = $this.find('button');
+    var $errorMsg = $this.find('.error-msg');
+    var $successMsg = $this.find('.success-msg');
+    var url = $form.attr('action');
 
-    $(document).ready(function(){
+    $input.on('keyup', function() {
+      if (!emailRegex.test($input.val())) {
+        $btn.attr('disabled', true);
+      } else {
+        $btn.attr('disabled', false);
+      }
+    });
 
-        // On the home page, move the blog icon inside the header 
-        // for better relative/absolute positioning.
+    $form.on('submit', function(e) {
+      e.preventDefault();
 
-        //$("#blog-logo").prependTo("#site-head-content");
+      $btn.attr('disabled', true);
+      if (emailRegex.test($input.val())) {
+        $input.attr('disabled', true);
+        $errorMsg.hide();
+        $successMsg.hide();
 
+        invite(url, $input.val()).then(
+          function() {
+            $input.attr('disabled', false);
+            $input.val('');
+            $successMsg.show();
+          },
+          function() {
+            $input.attr('disabled', false);
+            $btn.attr('disabled', false);
+            $errorMsg.show();
+          }
+        );
+      }
     });
+  }
 
-}(jQuery));
\ No newline at end of file
+  $.each($forms, setupForm);
+})(jQuery);
```

Comment:
The alternative is introduce react to the blog, figure out a way to partially compile both the JS and the SCSS (which depends on variables and mixins and so on), add it to the blog built (because the blog has no webpack), keep them in sync, etc. Does not sound much better to me, and this is way simpler.